[2023-07-28T13:35:32.033+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: toronto_real_estate_etl.Extract manual__2023-07-28T13:35:31.501823+00:00 [queued]>
[2023-07-28T13:35:32.040+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: toronto_real_estate_etl.Extract manual__2023-07-28T13:35:31.501823+00:00 [queued]>
[2023-07-28T13:35:32.040+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 1
[2023-07-28T13:35:32.051+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): Extract> on 2023-07-28 13:35:31.501823+00:00
[2023-07-28T13:35:32.055+0000] {standard_task_runner.py:57} INFO - Started process 5129 to run task
[2023-07-28T13:35:32.058+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'toronto_real_estate_etl', 'Extract', 'manual__2023-07-28T13:35:31.501823+00:00', '--job-id', '17', '--raw', '--subdir', 'DAGS_FOLDER/etl.py', '--cfg-path', '/tmp/tmpwvukmult']
[2023-07-28T13:35:32.059+0000] {standard_task_runner.py:85} INFO - Job 17: Subtask Extract
[2023-07-28T13:35:32.093+0000] {task_command.py:410} INFO - Running <TaskInstance: toronto_real_estate_etl.Extract manual__2023-07-28T13:35:31.501823+00:00 [running]> on host 7d0ee807347f
[2023-07-28T13:35:32.145+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='toronto_real_estate_etl' AIRFLOW_CTX_TASK_ID='Extract' AIRFLOW_CTX_EXECUTION_DATE='2023-07-28T13:35:31.501823+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-07-28T13:35:31.501823+00:00'
[2023-07-28T13:35:32.146+0000] {main.py:30} INFO - https://www.kijiji.ca/b-apartments-condos/page-1/c37l1700273
[2023-07-28T13:39:28.704+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/src/main.py", line 69, in extract
    e.getListings(2)
  File "/opt/airflow/dags/src/main.py", line 50, in getListings
    self.saveToJson(i)
  File "/opt/airflow/dags/src/main.py", line 56, in saveToJson
    filename = dir+'result_page_number_'+str(page_number)+"_"+timestr+".json"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
[2023-07-28T13:39:28.713+0000] {taskinstance.py:1350} INFO - Marking task as FAILED. dag_id=toronto_real_estate_etl, task_id=Extract, execution_date=20230728T133531, start_date=20230728T133532, end_date=20230728T133928
[2023-07-28T13:39:28.723+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 17 for task Extract (unsupported operand type(s) for +: 'NoneType' and 'str'; 5129)
[2023-07-28T13:39:28.768+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-07-28T13:39:28.788+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
